ebos/ecloutputblackoilmodule.hh

```diff
@@ -1044,170 +1089,277 @@ public:
-        ScalarBuffer regPressurePvHydrocarbon = computeFipForRegions_(pressureTimesHydrocarbonVolume_, fipnum_, ntFip);
-
-        ScalarBuffer fieldPressurePv = computeFipForRegions_(regPressurePv, fieldNum, 1, comunicateSum);
-        ScalarBuffer fieldPvHydrocarbon = computeFipForRegions_(regPvHydrocarbon, fieldNum, 1, comunicateSum);
-        ScalarBuffer fieldPressurePvHydrocarbon = computeFipForRegions_(regPressurePvHydrocarbon, fieldNum, 1, comunicateSum);
-
-        // output on io rank
-        // the original Fip values are stored on the first step
-        // TODO: Store initial Fip in the init file and restore them
-        // and use them here.
+        rsum.fieldPressurePv = sum(rsum.regPressurePv);
+        rsum.fieldPvHydrocarbon = sum(rsum.regPvHydrocarbon);
+        rsum.fieldPressurePvHydrocarbon = sum(rsum.regPressurePvHydrocarbon);
+        for (int fip_type = 0; fip_type < FipDataType::numFipTypes; fip_type++) {
+            const auto& regionFip = rsum.regFipValues[fip_type];
+            rsum.fieldFipValues[fip_type] = sum(regionFip);
+        }
+
+        if (is_fipnum) {
+            // The first time the outputFipLog function is run we store the inplace values in
+            // the initialInplace_ member. This has at least two problems:
+            //
+            //   o We really want the *initial* value - now we get the value after
+            //     the first timestep.
+            //
+            //   o For restarted runs this is obviously wrong.
+            //
+            // Finally it is of course not desirable to mutate state in an output
+            // routine.
+            if (!this->regionInitialInplace_) {
+                this->regionInitialInplace_.emplace();
+                for (int fip_type = 0; fip_type < FipDataType::numFipTypes; fip_type++)
+                    this->regionInitialInplace_.value()[fip_type] = rsum.regFipValues[fip_type];
+            }
+
+            if (!this->fieldInitialInplace_)
+                this->fieldInitialInplace_ = rsum.fieldFipValues;
+        }
+        return rsum;
+    }
+
+
+    std::unordered_map<std::string, RegionSum> makeRegionSums() {
+        std::unordered_map<std::string, RegionSum> rsum_map;
         const Opm::SummaryConfig summaryConfig = simulator_.vanguard().summaryConfig();
-        if (isIORank_()) {
-            // Field summary output
-            for (int i = 0; i<FipDataType::numFipValues; i++) {
-                std::string key = "F" + fipEnumToString_(i);
+
+        rsum_map.insert(std::make_pair("FIPNUM", makeRegionSum(this->regions_.at("FIPNUM"), true)));
+        for (const auto& fip_region : summaryConfig.fip_regions()) {
+            if (fip_region != "FIPNUM") {
+                const auto& region = this->regions_.at(fip_region);
+                RegionSum rsum = makeRegionSum(region, fip_region == "FIPNUM");
+                rsum_map.insert(std::make_pair(fip_region, std::move(rsum)));
+            }
+        }
```

Comment:
```C++
if (fip_region == "FIPNUM")
    continue;

rsum_map.emplace(fip_region, makeRegionSum(region, false));
```
This saves a level of indentation and automatically activates move semantics.

Reply:
OK